src/App.tsx

```diff
@@ -9,13 +9,32 @@ import {
 	Login,
 	Registration,
 } from 'components';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from 'store';
+import { getCourses } from 'store/courses/coursesSlice';
+import { getAuthors } from 'store/authors/authorsSlice';
+import { getAllAuthors, getAllCourses } from 'services';
 
 function App() {
+	const dispatch = useDispatch();
 	const navigate = useNavigate();
+	const token = useSelector((state: RootState) => state.user.token);
 
 	useEffect(() => {
-		const token = localStorage.getItem('token');
+		getAllCourses().then((response) => {
```

Comment:
this logic is related to business data, so it shouldn't be inside component logic. you have https://redux-toolkit.js.org/api/createAsyncThunk, so use it to create API calls, and place it near the related slice folder
there must be only dispatch calls like dispatch(fetchAllCourses) and dispatch(fetchAllAuthors)

Reply:
also, a note about ignoring eslint for empty dependency array for useEffect to navigating. it's okay to pass navigate function as a dependency, so you can remove ignoring the rule line and add navigate as a dependency

Reply:
I will move that logic to thunk in my next homework(task 4 - rewrite API to thunk)

Reply:
about navigation in useEffect dependency list, I made some small research and found the next solution: 
- no add navigate to [], because new navigation triggered when location changed, useHistory doesn't have that problem.
- rewrite to createBrowserRouter and use router than useNavigation (https://github.com/remix-run/react-router/issues/7634 - a few last comments in this discussion)

Tell me how to do better?

Reply:
than lfet it as is